### PR TITLE
standardize the notation for optional parameters

### PR DESCRIPTION
**Pull request by kartikeyakirar**
fixes https://github.com/insightsengineering/teal.modules.general/issues/674

### CODE REVIEW

File: R/tm_a_regression.R [hunk 1]
```diff
@@ -14,8 +14,8 @@
 #' Regressor variables from an incoming dataset with filtering and selecting.
 #' @param response (`data_extract_spec` or `list` of multiple `data_extract_spec`)
 #' Response variables from an incoming dataset with filtering and selecting.
-#' @param default_outlier_label (`character`, optional) The default column selected to label outliers.
-#' @param default_plot_type (`numeric`, optional) Defaults to Response vs Regressor.
+#' @param default_outlier_label (`character`) optional default column selected to label outliers.
+#' @param default_plot_type (`numeric`) optional defaults to "Response vs Regressor".
```
Comment (m7pr) [inline R/tm_a_regression.R:18]: I'm just guessing - should that be in backticks?
```suggestion
#' @param default_plot_type (`numeric`) optional defaults to `"Response vs Regressor"`.
```

File: R/tm_a_pca.R [hunk 2]
```diff
@@ -8,8 +8,8 @@
 #' @inheritParams teal::module
 #' @inheritParams shared_params
 #' @param dat (`data_extract_spec` or `list` of multiple `data_extract_spec`)
-#'   specifying columns used to compute PCA.
-#' @param font_size (`numeric`, optional) Specifies font size.
+#' specifying columns used to compute PCA.
+#' @param font_size (`numeric`) optional specifies font size.
```
Comment (vedhav) [inline R/tm_a_pca.R:12]:
```suggestion
#' @param font_size (`numeric`) optionally specifies font size.
```
Perhaps optionally would be better to use over here.

Comment (m7pr) [reply R/tm_a_pca.R:12]: I think that 

```r
(`numeric`) (optional) specifies font size.
```

would be the most natural in this situation : D

Comment (m7pr) [reply R/tm_a_pca.R:12]: or maybe 

```r
(`numeric`) optional, specifies font size.
```

Comment (vedhav) [reply R/tm_a_pca.R:12]: I like the second version from Marcin

Comment (kartikeyakirar) [reply R/tm_a_pca.R:12]: I believe we should standardize this and come to a decision. Do we incorporate it into the sentence directly, or should we use a specific notation? 


 Regarding the use of optionally or (option) create more formats so I think we should finalize it here first. https://github.com/insightsengineering/nestdevs-tasks/issues/52.

Comment (m7pr) [reply R/tm_a_pca.R:12]: just put a `,` after optional, so the sentence have logical sense. and it's all good.

**Comment by chlebowa** [comment]
Just a quick note. The fact that an argument is optional is readily apparent in the function definition (no default value) so there is no need to use the word ever time. We don't always state the default value in the argument description, after all, only when it matters.

File: R/tm_a_regression.R (hunk 1 above)
Comment (kartikeyakirar) [reply R/tm_a_regression.R:18]: its numeric type  and `Response vs Regressor` is label not choices that why I did not put backtick
image

**Comment by m7pr** [comment]
If we are sure that the parameter does not get set inside functions body to anything (if it's NULL/empty during the call) then maybe it's a good idea to remove `optional` completely.

**Comment by kartikeyakirar** [comment]
> The fact that an argument is optional is readily apparent in the function definition (no default value) 

I am confused,  Arguments are optional when you do not have to specify a value for them. They can have a default value, which is used if you do not specify.

**Comment by chlebowa** [comment]
@kartikeyakirar You are correct, my brain flipped the logic.

File: R/tm_a_pca.R (hunk 2 above)
Comment (kartikeyakirar) [reply R/tm_a_pca.R]: [c9ccb2c](https://github.com/insightsengineering/teal.modules.general/pull/698/commits/c9ccb2ce4224ac33291abbcedc7611152257fc30)

File: R/tm_a_regression.R (hunk 1 above)
Comment (m7pr) [reply R/tm_a_regression.R]: ok

**Review by m7pr — APPROVED** [review]

**Pull request merged**